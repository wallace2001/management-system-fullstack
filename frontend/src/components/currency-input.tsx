'use client';

import { Input } from '@/components/ui/input';
import { useEffect, useState } from 'react';

type CurrencyInputProps = {
  value: number;
  onChange: (value: number) => void;
  placeholder?: string;
};

export function CurrencyInput({ value, onChange, placeholder }: CurrencyInputProps) {
  const [formatted, setFormatted] = useState('');

  useEffect(() => {
    const formattedValue = new Intl.NumberFormat('pt-BR', {
      style: 'currency',
      currency: 'BRL',
    }).format(value);
    setFormatted(formattedValue);
  }, [value]);

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const rawValue = e.target.value.replace(/\D/g, '');
    const numericValue = Number(rawValue) / 100;
    onChange(numericValue);
  };

  return (
    <Input
      inputMode="numeric"
      value={formatted}
      onChange={handleChange}
      placeholder={placeholder || 'R$ 0,00'}
    />
  );
}
