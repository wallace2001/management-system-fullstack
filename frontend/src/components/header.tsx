'use client';

import { usePathname } from 'next/navigation';
import { Button } from '@/components/ui/button';
import { Sheet, SheetContent, SheetTitle, SheetTrigger } from '@/components/ui/sheet';
import { UserButton } from '@/modules/auth/components/user-button';
import { Menu } from 'lucide-react';
import { User } from 'next-auth';
import Link from 'next/link';

export default function Header({ user }: { user: User }) {
  const pathname = usePathname();

  return (
    <header className="bg-background fixed top-0 z-50 w-full border-b px-4 sm:px-8">
      <div className="mx-auto flex h-20 max-w-screen-xl items-center justify-between">
        <p className="text-2xl font-bold">
          <Link href="/">Management</Link>
        </p>

        <div className="hidden items-center gap-2 md:flex">
          <Button
            variant={pathname === '/products' ? 'default' : 'outline'}
            asChild
          >
            <Link href="/products">Produtos</Link>
          </Button>

          <Button
            variant={pathname === '/orders' ? 'default' : 'outline'}
            asChild
          >
            <Link href="/orders">Pedidos</Link>
          </Button>

          {user?.id && (
            <Button
              variant={pathname === '/users' ? 'default' : 'outline'}
              asChild
            >
              <Link href="/users">Usuários</Link>
            </Button>
          )}

          {user?.id ? <UserButton /> : (
            <Button asChild>
              <Link href="/auth">Entrar</Link>
            </Button>
          )}
        </div>

        <div className="md:hidden">
          <Sheet>
            <SheetTrigger asChild>
              <Button variant="outline" size="icon">
                <Menu className="h-5 w-5" />
              </Button>
            </SheetTrigger>
            <SheetContent side="right" className="w-64 p-6 space-y-4">
              <SheetTitle />
              <div className="space-y-2 pt-10">
                <Button
                  variant={pathname === '/products' ? 'default' : 'ghost'}
                  className="w-full justify-start"
                  asChild
                >
                  <Link href="/products">Produtos</Link>
                </Button>

                <Button
                  variant={pathname === '/orders' ? 'default' : 'ghost'}
                  className="w-full justify-start"
                  asChild
                >
                  <Link href="/orders">Pedidos</Link>
                </Button>

                {user?.id && (
                  <Button
                    variant={pathname === '/users' ? 'default' : 'ghost'}
                    className="w-full justify-start"
                    asChild
                  >
                    <Link href="/users">Usuários</Link>
                  </Button>
                )}

                {user?.id ? (
                  <div className="pt-4">
                    <UserButton />
                  </div>
                ) : (
                  <Button variant="default" className="w-full" asChild>
                    <Link href="/auth">Entrar</Link>
                  </Button>
                )}
              </div>
            </SheetContent>
          </Sheet>
        </div>
      </div>
    </header>
  );
}
