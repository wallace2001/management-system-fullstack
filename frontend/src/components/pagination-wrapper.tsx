'use client';

import {
  Pagination,
  PaginationContent,
  PaginationItem,
  PaginationPrevious,
  PaginationNext,
  PaginationLink,
  PaginationEllipsis,
} from '@/components/ui/pagination';

type Props = {
  currentPage: number;
  totalPages: number;
  onPageChange: (page: number) => void;
};

export function PaginationWrapper({
  currentPage,
  totalPages,
  onPageChange,
}: Props) {
  const pages = getPages(currentPage, totalPages);

  return (
    <Pagination className="mt-4">
      <PaginationContent>
        <PaginationItem>
          <PaginationPrevious
            onClick={() => currentPage > 1 && onPageChange(currentPage - 1)}
          />
        </PaginationItem>

        {pages.map((page, idx) => (
          <PaginationItem key={idx}>
            {page === '...' ? (
              <PaginationEllipsis />
            ) : (
              <PaginationLink
                isActive={page === currentPage}
                onClick={() => onPageChange(Number(page))}
              >
                {page}
              </PaginationLink>
            )}
          </PaginationItem>
        ))}

        <PaginationItem>
          <PaginationNext
            onClick={() =>
              currentPage < totalPages && onPageChange(currentPage + 1)
            }
          />
        </PaginationItem>
      </PaginationContent>
    </Pagination>
  );
}

function getPages(current: number, total: number): (number | string)[] {
  const delta = 2;
  const pages: (number | string)[] = [];

  for (let i = 1; i <= total; i++) {
    if (
      i === 1 ||
      i === total ||
      (i >= current - delta && i <= current + delta)
    ) {
      pages.push(i);
    } else if (pages[pages.length - 1] !== '...') {
      pages.push('...');
    }
  }

  return pages;
}
