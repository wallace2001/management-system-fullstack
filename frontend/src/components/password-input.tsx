'use client';

import { Input } from '@/components/ui/input';
import { cn } from '@/lib/utils';
import { Eye, EyeOff } from 'lucide-react';
import { useState } from 'react';

type PasswordInputProps = React.InputHTMLAttributes<HTMLInputElement>;

export function PasswordInput({ className, ...props }: PasswordInputProps) {
  const [showPassword, setShowPassword] = useState(false);

  return (
    <div className="relative">
      <Input
        type={showPassword ? 'text' : 'password'}
        className={cn('py-6 pr-10', className)}
        {...props}
      />
      <button
        type="button"
        onClick={() => setShowPassword((prev) => !prev)}
        className="absolute inset-y-0 right-3 flex items-center"
      >
        {showPassword ? (
          <EyeOff className="h-5 w-5 text-gray-600" />
        ) : (
          <Eye className="h-5 w-5 text-gray-600" />
        )}
      </button>
    </div>
  );
}
