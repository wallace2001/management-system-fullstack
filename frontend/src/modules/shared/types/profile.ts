export type Profile = {
  id: string;
  username: string;
  role: 'ADMIN' | 'USER';
  createdAt: Date;
};
