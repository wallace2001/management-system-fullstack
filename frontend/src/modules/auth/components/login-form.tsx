'use client';

import { zodResolver } from '@hookform/resolvers/zod';
import { useForm } from 'react-hook-form';
import { Input } from '@/components/ui/input';
import {
  Form,
  FormItem,
  FormControl,
  FormLabel,
  FormMessage,
  FormField,
} from '@/components/ui/form';
import { PasswordInput } from '@/components/password-input';
import { LoginFormValues, loginSchema } from '../schemas/login-schema';
import { SubmitButton } from '@/components/submit-button';
import Link from 'next/link';
import { useLoginMutation } from '../hooks/use-login-mutation';

export function LoginForm() {
  const loginMutation = useLoginMutation();
  const form = useForm<LoginFormValues>({
    resolver: zodResolver(loginSchema),
    defaultValues: {
      username: '',
      password: '',
    },
  });

  const onSubmit = (values: LoginFormValues) => {
    loginMutation.mutate(values);
  };

  return (
    <Form {...form}>
      <form
        onSubmit={form.handleSubmit(onSubmit)}
        className="space-y-6 bg-gray-50 px-6 py-8 sm:px-12"
      >
        <FormField
          control={form.control}
          name="username"
          render={({ field }) => (
            <FormItem>
              <FormLabel>Usuário</FormLabel>
              <FormControl>
                <Input className="h-12" placeholder="Seu usuário" {...field} />
              </FormControl>
              <FormMessage />
            </FormItem>
          )}
        />

        <FormField
          control={form.control}
          name="password"
          render={({ field }) => (
            <FormItem>
              <FormLabel>Senha</FormLabel>
              <FormControl>
                <PasswordInput placeholder="Sua senha" {...field} />
              </FormControl>
              <FormMessage />
            </FormItem>
          )}
        />

        <SubmitButton>Entrar</SubmitButton>
        <p className="text-center text-sm text-gray-600">
          {'Ainda não tem uma conta? '}
          <Link href="/auth/register" className="font-semibold text-gray-800">
            Cadastrar
          </Link>
        </p>
      </form>
    </Form>
  );
}
