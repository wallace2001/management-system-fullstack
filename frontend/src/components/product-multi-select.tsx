/* eslint-disable react-hooks/exhaustive-deps */
'use client';

import { cn } from '@/lib/utils';
import {
  Select,
  SelectContent,
  SelectItem,
  SelectTrigger,
  SelectValue,
} from '@/components/ui/select';
import { Input } from '@/components/ui/input';
import { useEffect, useState } from 'react';
import { Product } from '@/modules/shared/types/product';
import { useProductsQuery } from '@/modules/products/hooks/use-get-products';

type ProductMultiSelectProps = {
  value: Record<string, number>;
  onChange: (value: Record<string, number>) => void;
} & React.HTMLAttributes<HTMLDivElement>;

export function ProductMultiSelect({
  value,
  onChange,
  className,
  ...props
}: ProductMultiSelectProps) {
  const { data: products = [] } = useProductsQuery();
  const [selectedId, setSelectedId] = useState('');
  const [quantity, setQuantity] = useState(1);

  useEffect(() => {
    if (!value || typeof value !== 'object') {
      onChange({});
    }
  }, []);

  function handleAddProduct() {
    if (!selectedId || quantity < 1) return;

    const updated = {
      ...value,
      [selectedId]: (value[selectedId] || 0) + quantity,
    };

    onChange(updated);
    setSelectedId('');
    setQuantity(1);
  }

  function handleRemoveProduct(productId: string) {
    const updated = { ...value };
    delete updated[productId];
    onChange(updated);
  }

  return (
    <div className={cn('space-y-2', className)} {...props}>
      <div className="flex gap-2 items-center">
        <Select value={selectedId} onValueChange={setSelectedId}>
          <SelectTrigger className="w-full">
            <SelectValue placeholder="Selecione um produto" />
          </SelectTrigger>
          <SelectContent>
            {products.map((product: Product) => (
              <SelectItem key={product.id} value={product.id}>
                {product.name}
              </SelectItem>
            ))}
          </SelectContent>
        </Select>

        <Input
          type="number"
          min={1}
          value={quantity}
          onChange={(e) => setQuantity(Number(e.target.value))}
          className="w-24"
          placeholder="Qtd"
        />

        <button
          type="button"
          onClick={handleAddProduct}
          className="px-4 py-2 bg-primary text-white text-sm rounded"
        >
          Adicionar
        </button>
      </div>

      <div className="text-sm text-muted-foreground space-y-1">
        {Object.entries(value).length > 0 ? (
          Object.entries(value).map(([id, qty]) => {
            const name = products.find((p) => p.id === id)?.name || id;
            return (
              <div
                key={id}
                className="flex items-center justify-between border px-2 py-1 rounded"
              >
                <span>
                  {name} â€” <strong>{qty}x</strong>
                </span>
                <button
                  type="button"
                  className="text-xs text-red-500 hover:underline"
                  onClick={() => handleRemoveProduct(id)}
                >
                  Remover
                </button>
              </div>
            );
          })
        ) : (
          <p>Nenhum produto adicionado.</p>
        )}
      </div>
    </div>
  );
}
