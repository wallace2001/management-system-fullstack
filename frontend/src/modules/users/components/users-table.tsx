'use client';

import { Button } from '@/components/ui/button';
import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableHeader,
  TableRow,
} from '@/components/ui/table';
import { Skeleton } from '@/components/ui/skeleton';
import { Profile } from '@/modules/shared/types/profile';
import { Trash2, Pencil } from 'lucide-react';
import { useUserModals } from '../store/use-users-store';

export const UsersTable = ({
  users,
  isLoading = false,
}: {
  users?: Profile[];
  isLoading?: boolean;
}) => {
  const { openEdit, openDelete } = useUserModals();

  const renderSkeletonRows = () => {
    return Array.from({ length: 5 }).map((_, index) => (
      <TableRow key={`skeleton-${index}`}>
        <TableCell>
          <Skeleton className="h-4 w-24" />
        </TableCell>
        <TableCell>
          <Skeleton className="h-4 w-16" />
        </TableCell>
        <TableCell>
          <Skeleton className="h-4 w-20" />
        </TableCell>
        <TableCell className="space-x-2 text-right">
          <Skeleton className="inline-block h-8 w-8" />
          <Skeleton className="inline-block h-8 w-8" />
        </TableCell>
      </TableRow>
    ));
  };

  return (
    <Table>
      <TableHeader>
        <TableRow>
          <TableHead>Usuário</TableHead>
          <TableHead>Função</TableHead>
          <TableHead>Data de Criação</TableHead>
          <TableHead className="text-right">Ações</TableHead>
        </TableRow>
      </TableHeader>
      <TableBody>
        {isLoading
          ? renderSkeletonRows()
          : users?.map((user) => (
              <TableRow key={user.id}>
                <TableCell>{user.username}</TableCell>
                <TableCell>{user.role}</TableCell>
                <TableCell>
                  {new Date(user.createdAt).toLocaleDateString()}
                </TableCell>
                <TableCell className="space-x-2 text-right">
                  <Button
                    variant="outline"
                    size="sm"
                    onClick={() => openEdit(user)}
                  >
                    <Pencil className="h-4 w-4" />
                  </Button>
                  <Button
                    variant="destructive"
                    size="sm"
                    onClick={() => openDelete(user)}
                  >
                    <Trash2 className="h-4 w-4" />
                  </Button>
                </TableCell>
              </TableRow>
            ))}
      </TableBody>
    </Table>
  );
};
