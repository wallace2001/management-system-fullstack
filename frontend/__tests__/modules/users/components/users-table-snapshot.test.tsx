import { render, screen } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { UsersTable } from '@/modules/users/components/users-table';
import { Profile } from '@/modules/shared/types/profile';

const renderWithClient = (ui: React.ReactNode) => {
  const queryClient = new QueryClient();
  return render(
    <QueryClientProvider client={queryClient}>{ui}</QueryClientProvider>,
  );
};

describe('UsersTable', () => {
  const mockUsers: Profile[] = [
    {
      id: '1',
      username: 'admin',
      role: 'ADMIN',
      createdAt: new Date('2023-01-01'),
    },
    {
      id: '2',
      username: 'user',
      role: 'USER',
      createdAt: new Date('2023-02-01'),
    },
  ];

  it('deve renderizar os dados da tabela corretamente', () => {
    renderWithClient(<UsersTable users={mockUsers} isLoading={false} />);

    expect(screen.getByText('admin')).toBeInTheDocument();
    expect(screen.getByText('user')).toBeInTheDocument();
    expect(screen.getAllByRole('button')).toHaveLength(4); // 2 actions por linha
  });

  it('deve renderizar skeletons enquanto estiver carregando', () => {
    renderWithClient(<UsersTable isLoading />);
    expect(screen.getAllByRole('row')).toHaveLength(6); // 1 header + 5 skeletons
  });

  it('snapshot da tabela', () => {
    const { container } = renderWithClient(
      <UsersTable users={mockUsers} isLoading={false} />,
    );
    expect(container).toMatchSnapshot();
  });
});
