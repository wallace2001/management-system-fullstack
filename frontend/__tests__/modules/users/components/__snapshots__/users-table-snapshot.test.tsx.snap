// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersTable snapshot da tabela 1`] = `
<div>
  <div
    class="relative w-full overflow-x-auto"
    data-slot="table-container"
  >
    <table
      class="w-full caption-bottom text-sm"
      data-slot="table"
    >
      <thead
        class="[&_tr]:border-b"
        data-slot="table-header"
      >
        <tr
          class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
          data-slot="table-row"
        >
          <th
            class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-head"
          >
            Usuário
          </th>
          <th
            class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-head"
          >
            Função
          </th>
          <th
            class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-head"
          >
            Data de Criação
          </th>
          <th
            class="text-foreground h-10 px-2 align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] text-right"
            data-slot="table-head"
          >
            Ações
          </th>
        </tr>
      </thead>
      <tbody
        class="[&_tr:last-child]:border-0"
        data-slot="table-body"
      >
        <tr
          class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
          data-slot="table-row"
        >
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            admin
          </td>
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            ADMIN
          </td>
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            12/31/2022
          </td>
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] space-x-2 text-right"
            data-slot="table-cell"
          >
            <button
              class="focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive inline-flex shrink-0 cursor-pointer items-center justify-center text-sm font-medium whitespace-nowrap transition-all outline-none focus-visible:ring-[3px] disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4 bg-background hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 border shadow-xs h-8 gap-1.5 rounded-md px-3 has-[>svg]:px-2.5"
              data-slot="button"
            >
              <svg
                class="lucide lucide-pencil h-4 w-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.174 6.812a1 1 0 0 0-3.986-3.987L3.842 16.174a2 2 0 0 0-.5.83l-1.321 4.352a.5.5 0 0 0 .623.622l4.353-1.32a2 2 0 0 0 .83-.497z"
                />
                <path
                  d="m15 5 4 4"
                />
              </svg>
            </button>
            <button
              class="focus-visible:border-ring aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive inline-flex shrink-0 cursor-pointer items-center justify-center text-sm font-medium whitespace-nowrap transition-all outline-none focus-visible:ring-[3px] disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4 bg-destructive hover:bg-destructive/90 focus-visible:ring-destructive/20 dark:focus-visible:ring-destructive/40 dark:bg-destructive/60 text-white shadow-xs h-8 gap-1.5 rounded-md px-3 has-[>svg]:px-2.5"
              data-slot="button"
            >
              <svg
                class="lucide lucide-trash2 h-4 w-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 6h18"
                />
                <path
                  d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                />
                <path
                  d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                />
                <line
                  x1="10"
                  x2="10"
                  y1="11"
                  y2="17"
                />
                <line
                  x1="14"
                  x2="14"
                  y1="11"
                  y2="17"
                />
              </svg>
            </button>
          </td>
        </tr>
        <tr
          class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
          data-slot="table-row"
        >
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            user
          </td>
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            USER
          </td>
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
            data-slot="table-cell"
          >
            1/31/2023
          </td>
          <td
            class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] space-x-2 text-right"
            data-slot="table-cell"
          >
            <button
              class="focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive inline-flex shrink-0 cursor-pointer items-center justify-center text-sm font-medium whitespace-nowrap transition-all outline-none focus-visible:ring-[3px] disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4 bg-background hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 border shadow-xs h-8 gap-1.5 rounded-md px-3 has-[>svg]:px-2.5"
              data-slot="button"
            >
              <svg
                class="lucide lucide-pencil h-4 w-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.174 6.812a1 1 0 0 0-3.986-3.987L3.842 16.174a2 2 0 0 0-.5.83l-1.321 4.352a.5.5 0 0 0 .623.622l4.353-1.32a2 2 0 0 0 .83-.497z"
                />
                <path
                  d="m15 5 4 4"
                />
              </svg>
            </button>
            <button
              class="focus-visible:border-ring aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive inline-flex shrink-0 cursor-pointer items-center justify-center text-sm font-medium whitespace-nowrap transition-all outline-none focus-visible:ring-[3px] disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4 bg-destructive hover:bg-destructive/90 focus-visible:ring-destructive/20 dark:focus-visible:ring-destructive/40 dark:bg-destructive/60 text-white shadow-xs h-8 gap-1.5 rounded-md px-3 has-[>svg]:px-2.5"
              data-slot="button"
            >
              <svg
                class="lucide lucide-trash2 h-4 w-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 6h18"
                />
                <path
                  d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                />
                <path
                  d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                />
                <line
                  x1="10"
                  x2="10"
                  y1="11"
                  y2="17"
                />
                <line
                  x1="14"
                  x2="14"
                  y1="11"
                  y2="17"
                />
              </svg>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
